frontend/src/main/webui/src/components/FileTree.jsx

```diff
@@ -0,0 +1,152 @@
+import React, {PropTypes} from 'react';
+import { Row, Col, Grid, Table } from 'react-bootstrap';
+import { toPairs } from '../utils/utils';
+import { connect } from 'react-redux'
+import { bindActionCreators } from 'redux'
+import actions from '../actions/actions';
+import moment from 'moment';
+import apiNames from '../GefAPI';
+
+var path = require('path');
+
+class FileTree extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            folderOpen: {},
+        };
+    }
+
+    handleFolderClick(child, isOpen, e) {
+        let folderOpen = this.state.folderOpen;
+        let volumeInternalPath = path.join(child.path, child.name);
+        folderOpen[volumeInternalPath] = isOpen;
+        this.setState({folderOpen});
+    }
+
+    humanSize(sz) {
+        let K = 1000, M = K*K, G = K*M, T = K*G;
+
+        if (sz < K) {
+            return [sz,'B '];
+        } else if (sz < M) {
+            return [(sz/K).toFixed(2), 'KB'];
+        } else if (sz < G) {
+            return [(sz/M).toFixed(2), 'MB'];
+        } else if (sz < T) {
+            return [(sz/G).toFixed(2), 'GB'];
+        } else {
+            return [(sz/T).toFixed(2), 'TB'];
+        }
+    }
+
+    renderFile(file, indentStyle) {
+        let volumeInternalPath = path.join(file.path, file.name);
+        let iconClass = "glyphicon-file";
+        let browseButton;
+        let downloadButton;
+        let b2dropButton;
+        let isContentVisible = false;
+        if (file.isFolder) {
+            iconClass = "glyphicon-folder-close";
+            browseButton = <button style={{width:20, background:'none', border:'none', fontSize:20, padding:0}} onClick={ (e) => this.handleFolderClick(file, true, e)}>+</button>
+            if (this.state.folderOpen[volumeInternalPath] != null) {
+                if (this.state.folderOpen[volumeInternalPath]) {
+                    iconClass = "glyphicon-folder-open";
+                    browseButton = <button style={{width:20, background:'none', border:'none', fontSize:20, padding:0}} onClick={ (e) => this.handleFolderClick(file, false, e)}>-</button>
+                }
+            }
+        } else {
+            let volumeFilePath;
+            if (file.path == "") {
+                volumeFilePath = ""
+            } else {
+                volumeFilePath = "/" + file.path;
+            }
+
+            downloadButton = <a href={apiNames.volumes + "/" + this.props.selectedVolumeID + volumeFilePath + "/" + file.name + "?content"}><span className="glyphicon glyphicon-download-alt" aria-hidden={true}/></a>
+            b2dropButton = <span className="glyphicon glyphicon-cloud-upload" aria-hidden={true}/>
+        }
+
+        if (this.state.folderOpen[file.path] != null) {
```

Comment:
unneeded: `if this.state.folderOpen[file.path]` is enough. Null or undefined values are falsy. https://developer.mozilla.org/en-US/docs/Glossary/Falsy